internal/services/disbursement_management_service.go

```diff
@@ -210,19 +220,88 @@ func (s *DisbursementManagementService) StartDisbursement(ctx context.Context, d
 			}
 		}
 
-		// 4. Update all correct payment status to `ready`
+		// 4. Check if there is enough balance from the distribution wallet for this disbursement along with any pending disbursements
+		rootAccount, err := s.horizonClient.AccountDetail(
+			horizonclient.AccountRequest{AccountID: distributionPubKey})
+		if err != nil {
+			return fmt.Errorf("error getting details for root account from horizon client: %w", err)
+		}
+
+		var distributionBalance float64
+		for _, b := range rootAccount.Balances {
+			if b.Asset.Code == disbursement.Asset.Code && b.Asset.Issuer == disbursement.Asset.Issuer {
+				distributionBalance, err = strconv.ParseFloat(b.Balance, 64)
+				if err != nil {
+					return fmt.Errorf("error converting Horizon distribution account balance %s into float: %w", b.Balance, err)
+				}
+			}
+		}
+
+		disbursementAmount, err := strconv.ParseFloat(disbursement.TotalAmount, 64)
+		if err != nil {
+			return fmt.Errorf(
+				"error converting total amount %s for disbursement id %s into float: %w",
+				disbursement.TotalAmount,
+				disbursementID,
+				err,
+			)
+		}
+
+		var totalPendingAmount float64 = 0.0
+		incompletePayments, err := s.models.Payment.GetAll(ctx, &data.QueryParams{
+			Filters: map[data.FilterKey]interface{}{
+				data.FilterKeyStatus: []data.PaymentStatus{
+					data.ReadyPaymentStatus,
+					data.PendingPaymentStatus,
+					data.PausedPaymentStatus,
+				},
```

Comment:
I recommend creating a method that returns the list of incomplete payment statuses. Otherwise when we add new status we will likely miss updating this filter.